Cargo fails to install on OS X 10.11 public beta

Following the installation instructions I get the following:

````
"/Users/andrewrynhard/Desktop/cargo/rustc/bin/rustc" -V
dyld: Library not loaded: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_driver-74fa456f.dylib
  Referenced from: /Users/andrewrynhard/Desktop/cargo/rustc/bin/rustc
  Reason: image not found
make: *** [cargo-x86_64-apple-darwin] Trace/BPT trap: 5
````

I tried exporting `DYLD_LIBRARY_PATH` and `rustc` works

````
export DYLD_LIBRARY_PATH=/Users/andrewrynhard/Desktop/cargo/rustc/lib/
./rustc/bin/rustc -V
rustc 1.3.0-nightly (7ea2674c7 2015-07-13)`
````
however `make` still fails with the above error. Changing 

````
define CARGO_TARGET
cargo-$(1): $$(CARGO)
        "$$(CFG_RUSTC) -V"
        "$$(CARGO)" --version
        $$(CARGO) build --target $(1) $$(OPT_FLAG) $$(VERBOSE_FLAG) $$(ARGS)

test-unit-$(1): $$(CARGO)
        @mkdir -p target/$(1)/cit
        $$(CARGO) test --target $(1) $$(VERBOSE_FLAG) $$(only)
ended
````

to

````
...
        $$(CFG_RUSTC) -V
        $$(CARGO) --version
        $$(CARGO) build --target $(1) $$(OPT_FLAG) $$(VERBOSE_FLAG) $$(ARGS)
...
````
seems to fix the issue, but now I am getting
````
failed to run custom build command for `libssh2-sys v0.1.26`
````

--- a/Makefile.in
+++ b/Makefile.in
@@ -82,8 +82,8 @@ all: $(foreach target,$(CFG_TARGET),cargo-$(target))
 
 define CARGO_TARGET
 cargo-$(1): $$(CARGO)
-	"$$(CFG_RUSTC)" -V
-	"$$(CARGO)" --version
+	$$(CFG_RUSTC) -V
+	$$(CARGO) --version
 	$$(CARGO) build --target $(1) $$(OPT_FLAG) $$(VERBOSE_FLAG) $$(ARGS)
 
 test-unit-$(1): $$(CARGO)
@@ -141,7 +141,7 @@ target/doc/cargo/index.html:
 
 $(DOC_DIR)/%.html: src/doc/%.md src/doc/header.html src/doc/footer.html
 	@mkdir -p $(@D)
-	"$(CFG_RUSTDOC)" $< -o $(@D) $(DOC_OPTS)
+	$(CFG_RUSTDOC) $< -o $(@D) $(DOC_OPTS)
 
 $(DOC_DIR)/%: src/doc/%
 	@mkdir -p $(@D)

